app/models/task_task_category.rb

```diff
@@ -0,0 +1,4 @@
+class TaskTaskCategory < ApplicationRecord
```

Comment:
add a uniqness validation, there shud be just on entry in join table for a specific pair of task and task_category, read https://boringrails.com/tips/rails-unique-scope